src/social-network-provider/twitter.com/utils/selector.ts

```diff
@@ -0,0 +1,54 @@
+import { LiveSelector } from '@holoflows/kit'
+import { regexMatch } from '../../../utils/utils'
+
+const querySelector = <T extends HTMLElement>(selector: string) => {
+    return new LiveSelector().querySelector<T>(selector).enableSingleMode()
+}
+
+const querySelectorAll = <T extends HTMLElement>(selector: string) => {
+    return new LiveSelector().querySelectorAll<T>(selector)
+}
+
+/**
+ * @naming
+ * 御坂       (@Misaka_xxxx)
+ * userName   screenName
+ */
+
+export const bioQueryString = '[href*="header_photo"] + div [data-testid="UserDescription"]'
+
+export const bioCard = () => querySelector('[href*="header_photo"] + div')
+export const postViewMain = () =>
+    querySelector<HTMLElement>('[role="progressbar"] + div + div > div > div > div:first-of-type')
+
+const newPostEditor = '[role="main"] [role="progressbar"] ~ div'
+export const newPostEditorBelow = () => querySelector<HTMLDivElement>(`${newPostEditor} > div`)
+export const newPostEditorSelector = () => querySelector<HTMLDivElement>(`${newPostEditor} .DraftEditor-root`)
+export const newPostEditorInnerSelector = () =>
+    newPostEditorSelector().querySelector<HTMLDivElement>('.DraftEditor-editorContainer > div')
+
+export const editProfileButtonSelector = () =>
+    querySelector<HTMLAnchorElement>('[data-testid="primaryColumn"] [href="/settings/profile"]')
+export const editProfileTextareaSelector = () => querySelector<HTMLTextAreaElement>('textarea[placeholder*="bio"]')
+
+export const postsRootSelector = () => querySelector<HTMLElement>(`[data-testid="primaryColumn"] section`)
+
+export const postPopupSelector = () => querySelector('[aria-labelledby="modal-header"]')
+export const postsSelectors = () => querySelectorAll('article')
+export const postsContentSelectors = () => postsSelectors().querySelectorAll<HTMLElement>(`[lang]`)
+export const fromPostSelectorsSelectPostContentString = '[data-testid="tweet"] > div:nth-of-type(2)'
+
+// self infos
+const base = querySelector<HTMLScriptElement>('#react-root + script')
+const name = /"session":{.*?"user":{.*?"screen_name":"(.*?)","name":"(.*?)"}}/
```

Comment:
Uh.. Match by RegExp? It seems like a JSON or something...

Reply:
Match by RegExp could be faster than parsing the whole JSON in case I only want one value in it.